src/app_service/service/transaction/service.nim

```diff
@@ -368,28 +376,33 @@ QtObject:
           multiTxType: transactions.MultiTransactionType.MultiTransactionSend,
         ),
         paths,
-        common_utils.hashPassword(password),
+        password,
       )
 
-      if response.result{"hashes"} != nil:
-        for route in routes:
-          for hash in response.result["hashes"][$route.fromNetwork.chainID]:
-            self.watchTransaction(hash.getStr, from_addr, to_addr, $PendingTransactionTypeDto.WalletTransfer, " ", route.fromNetwork.chainID, track = false)
-            self.events.emit(SIGNAL_TRANSACTION_SENT, TransactionSentArgs(chainId: route.fromNetwork.chainID, txHash: hash.getStr, uuid: uuid , error: ""))
+      if password != "":
+        self.sendTransactionSentSignal(from_addr, to_addr, uuid, routes, response)
+
     except Exception as e:
-      self.events.emit(SIGNAL_TRANSACTION_SENT, TransactionSentArgs(chainId: 0, txHash: "", uuid: uuid, error: fmt"Error sending token transfer transaction: {e.msg}"))
+      self.sendTransactionSentSignal(from_addr, to_addr, uuid, @[], RpcResponse[JsonNode](), fmt"Error sending token transfer transaction: {e.msg}")
 
   proc transfer*(
     self: Service,
-    from_addr: string,
-    to_addr: string,
+    fromAddr: string,
+    toAddr: string,
     tokenSymbol: string,
     value: string,
     uuid: string,
     selectedRoutes: seq[TransactionPathDto],
     password: string,
-    sendType: SendType
+    sendType: SendType,
+    usePassword: bool,
+    doHashing: bool
   ) =
+    var finalPassword = ""
+    if usePassword:
+      finalPassword = password
+      if doHashing:
+        finalPassword = common_utils.hashPassword(password)
```

Comment:
It isn't the case right now, but it would perhaps be nice to have a comment/runtime check to ensure no plain password is ever by mistake sent over an RPC call, which I'm guessing is sniffable

Reply:
I am not sure that any rpc call uses password? Password is used only by status go, to verify/authorize/sign.